pkg/systemd/notifyproxy/notifyproxy.go

```diff
@@ -189,7 +188,16 @@ func (p *NotifyProxy) SocketPath() string {
 // Close closes the listener and removes the socket.
 func (p *NotifyProxy) Close() error {
 	defer os.Remove(p.socketPath)
-	return p.connection.Close()
+	if err := p.connection.Close(); err != nil {
+		// Listen has to continue run in the background and blocks on
+		// read, so both listen and Close() get the ErrClosed error
+		// (see #19175).
+		if errors.Is(err, net.ErrClosed) {
+			return nil
+		}
+		return err
+	}
```

Comment:
This seems wrong to me? To me this sounds like ErrClosed is happening because another caller already closed the connection? So should we not just fix that problem by not having two callers of close()?

I can at least see two duplicated callers of close in https://github.com/containers/podman/blob/7c0e926820881583b181985b31f7485e1a32d24c/pkg/domain/infra/abi/play.go#L916-L939

Reply:
There is even a third close() call here:
https://github.com/containers/podman/blob/7c0e926820881583b181985b31f7485e1a32d24c/pkg/domain/infra/abi/play.go#L180-L188

Reply:
I think you are right.  I tracked the error in the standard lib and was convinced it was due to the blocking read.